src/fdosecrets/objects/Collection.cpp

```diff
@@ -95,6 +98,20 @@ namespace FdoSecrets
         } else {
             cleanupConnections();
         }
+
+        return true;
+    }
+
+    QDBusAbstractAdaptor* Collection::createAdaptor()
+    {
+        return new CollectionAdaptor(this);
```

Comment:
This is a great way to leak memory.

Reply:
Adaptors are parented to `this` and will only be created once.

Reply:
I'm open to how to make the intent more clear

Reply:
Now the creation of the adaptor is enforced in the constructor.

Reply:
Use a QSharedPointer

Reply:
We only use the parent child relationship for QWidgets. It causes problems unless you use QPointer to reference the children because you scatter dangling pointers around when the parent deletes.

Reply:
I understand. In this particular case, the adaptor has to be the child otherwise Qt DBus won't find it. Anyway, this method is gone now and I create these adaptors directly in the DBusObjectHelper constructor.

As for others (Service, Collection, Item, Session), it's indeed possible and maybe more appropriate to move away from the parent child relationship. But let's save it for another PR.

Reply:
Sounds good to me